### PR TITLE
[Hub Generated] Review request for Microsoft.Web to add version 2018-02-01

### PR DESCRIPTION
**Pull request by dejo-msft**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/dejo-msft/azure-rest-api-specs/dev-web-Microsoft.Web-2018-02-01).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by praries880** [comment]
@dejo-msft Can you add examples for the new API you added in the PR?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Please add a description

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Given that endpoints and ports are separate arrays, how does this work in practice? Each endpoint IP address supports each port?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: What does this mean? Whether you are able to make an inbound TCP connection to the ipAddress + port? From where?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Fill out description

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: What does this mean? Please can you be more specific with the description?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: This is a similar model to "InboundEnvironmentEndpoint" except it doesn't include the ports. Is this intentional?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: inboundnetworkdependenciesendpoints feels quite long-winded and it's not obvious to me what this actually is. Let's have a discussion to go over some of the naming.

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

**Comment by praries880** [comment]
@dejo-msft Pinging back... are you working on the ARM feedback

**Comment by ravbhatnagar** [comment]
@dejo-msft - please let us know when its ready for ARM re-review.

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Unfortunately, this API is in production and in use by customers and the Azure Portal. I've created an issue here https://github.com/Azure/azure-rest-api-specs/issues/5255 which will require a new API version to fix.

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Unfortunately, as I mentioned before, this API is in production and the signature can't be changed in this version. I'll work on options in the API name.  Does dropping "Endpoints" make more sense? Initially, the api was a list of <IP, Port> but the requirements changed. InboundNetworkDependencies and OutboundNetworkDependencies?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Yes, this is intentional. Inbound endpoints intentionally leave out the Fully Qualified Domain Name to avoid including infrastructure implementation details. Outbound endpoints include the FQDN so that they can be whitelisted by a layer 7 firewall. We don't have the same support for inbound endpoints so we only have the IP Address and port.

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: This is a short description of what the endpoints is accessed for. E.G. "App Service management", "App Service Environment VIP", or "App Service Environment subnet".
Does this more accurately represent this? 
"Short description of why the endpoints are accessed."

**Comment by dejo-msft** [comment]
I've addressed feedback on the description texts and added examples of the API. 
@ravbhatnagar and @praries880 This is ready for ARM review again. Sorry for the delay.

**Comment by praries880** [comment]
@dejo-msft  kindly fix the [linter errors](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/499038791#L815)

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (KrisBash) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Understood. Would you please lowerCamelCase the type name? `inboundNetworkDependenciesEndpoints`

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (KrisBash) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: While not functionally impactful given case insensitivity in URIs, the resourceTypeNames should follow lowerCamelCase convention

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (KrisBash) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Nit - inconsistent casing of "Domain Name" in the EndpointDependency description vs domainName property description

**Review by KrisBash — APPROVED** [review]
Feedback looks to be addressed. i've added a few non-blocking comments regarding casing..

**Comment by dejo-msft** [comment]
@sergey-shandar you pushed a fix on https://github.com/Azure/azure-rest-api-specs/pull/5288 to the linterdiff tool. Can you do the same here?

**Comment by praries880** [comment]
@dejo-msft  kindly fix the CI issues, we have ARM signoff... We wont be able to merge this in till the CI is fixed.

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: Fixed these. Thanks!

**Review by praries880 — APPROVED** [review]

**Comment by praries880** [comment]
@antmarti-microsoft  can you kindly sign off the PR, i cant merge this until you sign off :)

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: The example uses paging, but this has not been called out here with an "x-ms-pageable" declaration. Please can you either update the example, or the schema, depending on whether or not you are using paging on this response?

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (antmarti-microsoft) [inline specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: The example uses paging, but this has not been called out here with an "x-ms-pageable" declaration. Please can you either update the example, or the schema, depending on whether or not you are using paging on this response?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]
Please address the discrepancy between examples (using paging), and the response declaration (not using paging).

File: specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json
Comment (dejo-msft) [reply specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/AppServiceEnvironments.json]: The APIs are not paged. I've removed them from examples. Thanks for catching this!

**Review by anthony-c-martin — APPROVED** [review]

**Pull request merged**
